pkg/serializer/internal/metrics/iterable_series.go

```diff
@@ -47,9 +72,396 @@ func (series IterableSeries) DescribeCurrentItem() string {
+					return err
+				}
+
+				return nil
+			})
+			if err != nil {
+				return err
+			}
+
+			err = ps.String(seriesMetric, serie.Name)
+			if err != nil {
+				return err
+			}
+
+			err = serie.Tags.ForEachErr(func(tag string) error {
+				return ps.String(seriesTags, tag)
+			})
+			if err != nil {
+				return err
+			}
+
+			err = ps.Int32(seriesType, serie.MType.SeriesAPIV2Enum())
+			if err != nil {
+				return err
+			}
+
+			err = ps.String(seriesSourceTypeName, serie.SourceTypeName)
+			if err != nil {
+				return err
+			}
+
+			err = ps.Int64(seriesInterval, serie.Interval)
+			if err != nil {
+				return err
+			}
+
+			// (Unit is omitted)
+
+			for _, p := range serie.Points {
+				err = ps.Embedded(seriesPoints, func(ps *molecule.ProtoStream) error {
+					err = ps.Int64(pointTimestamp, int64(p.Ts))
+					if err != nil {
+						return err
+					}
+
+					err = ps.Double(pointValue, p.Value)
+					if err != nil {
+						return err
+					}
+
+					return nil
+				})
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		if len(serie.Points) > maxPointsPerPayload {
+			// this series is just too big to fit in a payload (even alone)
+			err = stream.ErrItemTooBig
+		} else if pointsThisPayload+len(serie.Points) > maxPointsPerPayload {
+			// this series won't fit in this payload, but will fit in the next
+			err = stream.ErrPayloadFull
+		} else {
+			// Compress the protobuf metadata and the marshaled series
+			err = addToPayload()
+		}
+
+		switch err {
+		case stream.ErrPayloadFull:
+			expvarsPayloadFull.Add(1)
+			tlmPayloadFull.Inc()
+
+			err = finishPayload()
+			if err != nil {
+				return nil, err
+			}
+
+			err = startPayload()
+			if err != nil {
+				return nil, err
+			}
+
+			// Add it to the new compression buffer
+			err = addToPayload()
+			if err == stream.ErrItemTooBig {
+				// Item was too big, drop it
+				expvarsItemTooBig.Add(1)
+				tlmItemTooBig.Inc()
+				continue
+			}
+			if err != nil {
+				// Unexpected error bail out
+				expvarsUnexpectedItemDrops.Add(1)
+				tlmUnexpectedItemDrops.Inc()
+				return nil, err
+			}
+		case stream.ErrItemTooBig:
+			// Item was too big, drop it
+			expvarsItemTooBig.Add(1)
+			tlmItemTooBig.Add(1)
+		case nil:
+			continue
+		default:
+			// Unexpected error bail out
+			expvarsUnexpectedItemDrops.Add(1)
+			tlmUnexpectedItemDrops.Inc()
+			return nil, err
+		}
+	}
+
+	// if the last payload has any data, flush it
+	err = finishPayload()
+	if err != nil {
+		return nil, err
+	}
+
+	return payloads, nil
+}
+
+type serieIterator interface {
+	MoveNext() bool
+	Current() *metrics.Serie
+}
+
+// MarshalJSON serializes timeseries to JSON so it can be sent to V1 endpoints
+//FIXME(maxime): to be removed when v2 endpoints are available
+func (series IterableSeries) MarshalJSON() ([]byte, error) {
+	// use an alias to avoid infinite recursion while serializing a Series
+	type SeriesAlias Series
```

Comment:
Are we serializing Series to JSON anywhere now? Would it be possible to remove custom serializer and this alias now?

Reply:
Unfortunately, the Serverless Agent uses this method.